src/http_server.cc

```diff
@@ -3225,6 +3225,42 @@ HTTPAPIServer::HandleGenerate(
       req, RestrictedCategory::INFERENCE, restricted_apis_);
 
   AddContentTypeHeader(req, "application/json");
+#ifdef TRITON_ENABLE_METRICS
+  // logic to add kv_cache metrics to response header
+  // Get the metrics in Prometheus format
+  if (std::getenv("ORCA_HEADER_METRIC_TYPE") != nullptr) {
+    const std::string orca_type = std::getenv("ORCA_HEADER_METRIC_TYPE");
+    TRITONSERVER_Metrics* metrics = nullptr;
+    TRITONSERVER_Error* err =
+        TRITONSERVER_ServerMetrics(server_.get(), &metrics);
+    if (err == nullptr) {
+      const char* base;
+      size_t byte_size;
+      err = TRITONSERVER_MetricsFormatted(
+          metrics, TRITONSERVER_METRIC_PROMETHEUS, &base, &byte_size);
+      if (err == nullptr) {
+        std::string kv_utilization(base, byte_size);
+        // Extract the KV utilization metrics from the Prometheus formatted
+        // string.
+        std::string extracted_kv_metrics =
+            ExtractKVMetrics(kv_utilization, orca_type);
+        if (!extracted_kv_metrics.empty()) {
```

Comment:
can we print a log if metric is empty?

Reply:
We could, however this will always be the case if the server backend is not TensorRT-LLM. My goal was to make it unintrusive if the backend metrics weren't present since Triton can use different backends, however it could be argued that if the feature flag environment variable is set we are expecting the metrics to be present, and thus it should be fine to treat this as a warning state and print a log. I can add logging here similar to how I log an error if the environment variable is set to an invalid type.

Reply:
Can you add a log for feature enabled with non compatible environments "Ignoring ORCA_METRIC_FORMAT, backend X not supported" Only log this once, not part of this, at the entrypoint of server setup

Reply:
I'm not sure if there's a simple way to get the backend from the frontend without just checking the metrics and seeing if the tensorrt ones are present (there is a "model" parameter which could be set to TensorRT as that's what llama was provisioned with, but in many cases it will be hidden as "ensamble" as there are many layers to the model one of which is tensorrt-llm). I added such that if ORCA_METRIC_FORMAT is set but tensorrt metrics aren't available, it will log an error describing that, though it's at inference time when we get the metrics.